### PR TITLE
Added update road map step to release checklist

### PR DESCRIPTION
**Pull request by JonasHelming**
relates to #9232

Signed-off-by: Jonas Helming <jhelming@eclipsesource.com>

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: doc/Publishing.md [hunk 1]
```diff
@@ -147,3 +147,7 @@ To successfully complete a `yarn upgrade`, one must:
 
 * Close current release [milestone](https://github.com/eclipse-theia/theia/milestones).
 * Create the next two milestones in the case they do not already exist. Generally, the release is performed on the last Thursday of the month, but there may be exceptions (bug fix release, holidays, etc.)
+
+### Update Road Map
```
Comment (vince-fugnitto) [inline doc/Publishing.md:151]: I think we should update all occurrences, it is valid:

```suggestion
### Update Roadmap
```

**Review by vince-fugnitto — COMMENTED** [review]
@JonasHelming do we need to document maintaining the roadmap as part of the release procedure (which is generally performed by Paul or I). I believe the roadmap can be updated at any time, not necessarily after a release and not necessarily by those who release.

File: doc/Publishing.md [hunk 2]
```diff
@@ -147,3 +147,7 @@ To successfully complete a `yarn upgrade`, one must:
...
+* If the current release is the last in a quarter, ask team to update the [road map](https://github.com/eclipse-theia/theia/wiki/Roadmap) of the past quarter (close, remove or move items)
```
Comment (kittaakos) [inline doc/Publishing.md:152]: `ask team` -> `ask the team`.

Comment (kittaakos) [reply doc/Publishing.md:152]: Please replace `road map` -> `roadmap`.

File: doc/Publishing.md [hunk 3]
```diff
@@ -147,3 +147,7 @@ To successfully complete a `yarn upgrade`, one must:
...
+* If the current release is the second in a quarter, create [road map template](https://github.com/eclipse-theia/theia/wiki/Roadmap) for the next quarter, ask team to contribute to it and add it to the agenda of the Theia dev meeting
```
Comment (kittaakos) [inline doc/Publishing.md:153]: `ask team` -> `ask the team`.

**Review by kittaakos — APPROVED** [review]
Thank you! <3

**Comment by JonasHelming** [comment]
> @JonasHelming do we need to document maintaining the roadmap as part of the release procedure (which is generally performed by Paul or I). I believe the roadmap can be updated at any time, not necessarily after a release and not necessarily by those who release.

I quickly talked to Marc about this. I agree with you that it is not really a part of the release process and this step does not have to be done by the same people. This is more about having a trigger that this process step is executed, i.e. a reminder. To my knowledge, this is the only process that is regularly executed. For this reason I thought it is pragmatic to place the trigger here. As we (to my knowledge) have not any other processed like this, it might be overhead to create a completly new checklist just for this small step. If you want, I could clarify, that the roadmap update should be triggered by the project lead or somebody else?

**Review by vince-fugnitto — APPROVED** [review]
The changes look fine to me, we can always add to it, move it somewhere else in the future if necessary.

**Pull request merged**